### PR TITLE
feat: tree colors based on rain and water

### PR DESCRIPTION
**Pull request by Jaszkowic**
Update tree color based on sum of rain and water

- only affects trees age 5-10 years
- if rain+waterings < 100 then color = orange
- if rain+waterings >= 100 and < 200 then color = yellow
- if rain+waterings > 200 then color = green

for now, only works when no filter is active (because they are being reworked in another PR right now)

For testing, make sure to clear / disable cache to get the most up to date Mapbox layer.

### CODE REVIEW

File: src/components/tree-detail/tree-water-needs/watering-dialog.tsx [hunk 1]
```diff
@@ -52,6 +57,12 @@ export const WateringDialog: React.FC = () => {
 				// wait for the dialog to close before expanding the last waterings
 				setTimeout(() => {
 					setIsLastWateringsExpanded(true);
+
+					// Update the map with the new watering amount
+					const treeId = useTreeStore.getState().selectedTreeId;
+					if (treeId) {
+						addTodayWatering(treeId, amount);
+					}
```
Comment (raphael-arce) [inline src/components/tree-detail/tree-water-needs/watering-dialog.tsx:65]: is this located in the timeout by design, so the user can witness it change colors?

**Review by raphael-arce — CHANGES_REQUESTED** [review]
a few suggestions/questions, otherwise works very good so far, very nice feature! 👍

File: src/components/tree-detail/hooks/use-update-tree-waterings.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	const addTodayWatering = (treeId: string, amount: number) => {
+		updateTreeWaterings(treeId, amount, UpdateAction.ADD);
+	};
+
+	const removeTodayWatering = (treeId: string, amount: number) => {
+		updateTreeWaterings(treeId, amount, UpdateAction.REMOVE);
+	};
+
+	return { addTodayWatering, removeTodayWatering };
+}
```
[the remark this answers is not in the document]
Comment (Jaszkowic) [reply src/components/tree-detail/hooks/use-update-tree-waterings.tsx:39]: changed in latest commit

File: src/components/map/hooks/use-map-setup.tsx [hunk 3]
```diff
@@ -166,4 +168,39 @@ export function useMapSetup(
...
 	}, [mapContainer]);
+
+	const { todaysWaterings, loadTodaysWaterings } = useTreeStore();
+
+	useEffect(() => {
+		const loadData = async () => {
+			await loadTodaysWaterings();
+		};
+		loadData();
+	}, []);
```
[the remark this answers is not in the document]
Comment (Jaszkowic) [reply src/components/map/hooks/use-map-setup.tsx:179]: Would make sense, but I'm afraid that having a timer ticking could be tricking to get right with all edge cases that could happen in a React app. I'm open to it, but let's have it in a separate pull request if we decide to do it.

File: src/components/tree-detail/tree-water-needs/watering-dialog.tsx (hunk 1 above)
Comment (Jaszkowic) [reply src/components/tree-detail/tree-water-needs/watering-dialog.tsx:63]: yes, it's in the timeout for the user's satisfaction

File: src/components/map/hooks/use-map-setup.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (raphael-arce) [reply src/components/map/hooks/use-map-setup.tsx:179]: makes sense. My first idea would be to also load the watering data when the selectedTreeId changes. But let's make that in a separate PR 👍

**Review by raphael-arce — APPROVED** [review]
lgtm 👍

**Pull request merged**